blocks/library/image/index.js

```diff
@@ -72,6 +75,23 @@ registerBlock( 'core/image', {
 	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, alt, caption } = attributes;
 
+		if ( ! url ) {
+			return (
```

Comment:
That last example looks alright, but it opens too much for devs to handle, and differences will pop up. Empty could be just the comment (leaving blocks to implement more fine grained UI handling when specific attributes are missing), which could be a nice way to define a template:

```
<!-- wp:core/text --><!-- /wp:core/text -->
<!-- wp:core/image --><!-- /wp:core/image -->
<!-- wp:core/button --><!-- /wp:core/button -->
```

In any case I think forcing block authors to have a separate function for returning the empty render, regardless of whether it is called from `edit` manually, could be helpful.